### PR TITLE
feature - add overview page

### PR DESCRIPTION
**Pull request by ilinzweilin**
UI:
- create Pool OverView Page
- LoanList: remove column LoanOwner &  ME labels  -> refactor Loans & Overview container

ENV: 
- setup env to support multiple deployments
- create Pool env & type
- add pools & pool env validators
- update env & env.develop
- remove nftDataDefinitions

STATE:
- refactor analytics ducks: replace analytics with pool
- add total token supply for each tranche to the pool state

### CODE REVIEW

File: components/Loan/List/index.tsx [hunk 1]
```diff
@@ -33,25 +31,6 @@ class LoanList extends React.Component<Props> {
...
-            </Box>
-
-          </div>
-        },
-        {
-          header: '', property: '', align: 'end',
-          render: (l: Loan) => <div>
-            {proxies.includes(l.ownerOf.toString()) && <Badge text={'Me'} />}
-          </div>
-        },
```
Comment (philipstanislaus) [inline components/Loan/List/index.tsx:54]: Is this removal intended? How do users find their own loans, e. g. in the demo UI?

File: components/MintNFT/index.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ class MintNFT extends React.Component<Props, State> {
 
   mint = async () => {
     const { referenceId, assetType, amount, tokenId } = this.state;
-    const registry = config.contractAddresses.COLLATERAL_NFT;
+    const registry = config.pools[0].addresses.COLLATERAL_NFT;
```
Comment (philipstanislaus) [inline components/MintNFT/index.tsx:48]: Since `pool[0]` is hardcoded here – won't this imply that we might use the wrong registry if we are on another poll?

File: components/MintNFT/index.tsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ class MintNFT extends React.Component<Props, State> {
 
   render() {
     const { is, tokenId, errorMsg, referenceId, assetType, amount } = this.state;
-    const registry = config.contractAddresses.COLLATERAL_NFT;
+    const registry = config.pools[0].addresses.COLLATERAL_NFT;
```
Comment (philipstanislaus) [inline components/MintNFT/index.tsx:69]: Same comment

File: config.ts [hunk 4]
```diff
@@ -32,66 +27,52 @@ export interface DisplayedField {
...
-    }))
-  }),
-  displayedFields: yup.array(yup.object().shape({
-    key: yup.string().required().min(1),
-    label: yup.string().required().min(1),
-    type: yup.string().required().min(1),
-    decimals: yup.number().min(0),
-    precision: yup.number().min(0)
-  }))
+  COLLATERAL_NFT: yup.string().length(42).matches(/0x[0-9a-fA-F]{40}/),
```
Comment (philipstanislaus) [inline config.ts:45]: Is this address not required? If not, we should change the type to `string?` in line 9.

File: config.ts [hunk 5]
```diff
@@ -32,66 +27,52 @@ export interface DisplayedField {
...
+  DEPLOYMENT_NAME: yup.string().required()
```
Comment (philipstanislaus) [inline config.ts:46]: Why is a name part of the addresses?

File: config.ts [hunk 6]
```diff
@@ -32,66 +27,52 @@ export interface DisplayedField {
...
 });
 
 const contractConfigSchema = yup.object().required().shape({
   JUNIOR_OPERATOR: yup.mixed<'ALLOWANCE_OPERATOR'>().required().oneOf(['ALLOWANCE_OPERATOR']),
   SENIOR_OPERATOR: yup.mixed<'PROPORTIONAL_OPERATOR'>().required().oneOf(['PROPORTIONAL_OPERATOR'])
```
Comment (philipstanislaus) [inline config.ts:51]:
```suggestion
  SENIOR_OPERATOR: yup.mixed<'PROPORTIONAL_OPERATOR' | 'ALLOWANCE_OPERATOR'>().required().oneOf(['PROPORTIONAL_OPERATOR', 'ALLOWANCE_OPERATOR'])
```

File: config.ts [hunk 7]
```diff
@@ -32,66 +27,52 @@ export interface DisplayedField {
...
-  network: 'Mainnet' | 'Kovan';
-  contractConfig:  ContractConfig;
-}
+const poolSchema = yup.object().required().shape({
+  addresses: contractAddressesSchema,
+  graph: yup.string().required(),
+  contractConfig: contractConfigSchema,
+  name: yup.string().required(),
+  description: yup.string().required(),
+  slug: yup.string().required(),
```
Comment (philipstanislaus) [inline config.ts:60]: Do we need the slug? We are using the root address as the url param now, right?

File: config.ts [hunk 8]
```diff
@@ -32,66 +27,52 @@ export interface DisplayedField {
...
+  asset: yup.string().required()
```
Comment (philipstanislaus) [inline config.ts:61]: Can we call this `assetType` and supply a enum, like the following?

```suggestion
  asset: yup.mixed<'invoice' | 'bla'>().required().oneOf(['invoice', 'bla'])
```

File: pages/[root]/index.tsx [hunk 9]
```diff
@@ -1,32 +1,42 @@
...
+    );
+  }
 }
 
 export async function getStaticPaths() {
   // We'll pre-render only these paths at build time.
   const paths = [{
-    params: { root: config.contractAddresses.ROOT_CONTRACT }
+    params: { root: config.pools[0].addresses.ROOT_CONTRACT }
   }];
```
Comment (philipstanislaus) [inline pages/[root]/index.tsx:40]: The correct code here should be:
```
const paths = config.pools.map(pool => ({ params: { root: pool.addresses.ROOT_CONTRACT } }))
```

File: containers/Overview/index.tsx [hunk 10]
```diff
@@ -1,62 +1,152 @@
...
+            <Text margin={{ top: 'small' }}> {description} </Text>
+          </Box>
+        </Box>
+      </Box>
+
+      <Heading level="4" margin={{ top: 'xsmall' }}>Latest loans</Heading>
+      {(loans!.loansState === 'loading') ?
+        <Spinner height={'calc(100vh - 89px - 84px)'} message={'Loading...'} /> :
+        <LoanListData loans={latestLoans} userAddress={userAddress}> </LoanListData>
+      }
```
Comment (philipstanislaus) [inline containers/Overview/index.tsx:140]: Shall we have a pool loading spinner as well?

File: components/MintNFT/index.tsx (hunk 2 above)
Comment (ilinzweilin) [reply components/MintNFT/index.tsx:48]: yes, it is not ideal at all to hardcode it like this. 
The page is currently not exposed and is just used by us for testing purposes. I neeeded it to run for testing. I think we can also just hardcode the nftRegistry address in the code directly for now

File: config.ts (hunk 4 above)
Comment (ilinzweilin) [reply config.ts:45]: not required. Just needed for testing purposes & the old demo setup.

File: config.ts (hunk 5 above)
Comment (ilinzweilin) [reply config.ts:46]: let's take it out. I is autogenerated by the contract deployment and helps to distinguish between address json files, but definitely not needed in the ui - env

File: .travis.yml [hunk 11]
```diff
@@ -22,4 +22,4 @@ script:
...
+  - NEXT_PUBLIC_ETHERSCAN_URL=https://kovan.etherscan.io NEXT_PUBLIC_RPC_URL=https://kovan.infura.io/v3/092108ec6aea46ab97b2175b45130455 NEXT_PUBLIC_ENV=demo NEXT_PUBLIC_TRANSACTION_TIMEOUT=3600 NEXT_PUBLIC_TINLAKE_DATA_BACKEND_URL=https://api.thegraph.com/subgraphs/name/centrifuge/tinlake-kovan-staging NEXT_PUBLIC_POOLS=[ {"addresses":{ "DEPLOYMENT_NAME":"Kovan Naxos Deployment", "ROOT_CONTRACT":"0x41b7b379dee711b1a9bfbabd4b1309a584f5fe5a", "ACTIONS":"0x60cc4a2868559e513112f0742e6f546d339ea17b", "PROXY_REGISTRY":"0x99a31c86d95e907dcf0001857b8aa83456e746c1", "COLLATERAL_NFT": "0xac0c1ef395290288028a0a9fdfc8fdebebe54a24"}, "graph":"https://api.thegraph.com/subgraphs/name/centrifuge/tinlake-kovan-staging", "contractConfig":{"JUNIOR_OPERATOR":"ALLOWANCE_OPERATOR","SENIOR_OPERATOR":"PROPORTIONAL_OPERATOR"}, "name":"Kovan Naxos Deployment","description":"This is a test description", "slug":"testSlug", "asset":"invoice" } ]
```
Comment (philipstanislaus) [inline .travis.yml:25]:
```suggestion
  - NEXT_PUBLIC_ETHERSCAN_URL=https://kovan.etherscan.io NEXT_PUBLIC_RPC_URL=https://kovan.infura.io/v3/092108ec6aea46ab97b2175b45130455 NEXT_PUBLIC_ENV=demo NEXT_PUBLIC_TRANSACTION_TIMEOUT=3600 NEXT_PUBLIC_TINLAKE_DATA_BACKEND_URL=https://api.thegraph.com/subgraphs/name/centrifuge/tinlake-kovan-staging NEXT_PUBLIC_POOLS=[{"addresses":{"DEPLOYMENT_NAME":"Kovan Naxos Deployment","ROOT_CONTRACT":"0x41b7b379dee711b1a9bfbabd4b1309a584f5fe5a","ACTIONS":"0x60cc4a2868559e513112f0742e6f546d339ea17b","PROXY_REGISTRY":"0x99a31c86d95e907dcf0001857b8aa83456e746c1","COLLATERAL_NFT":"0xac0c1ef395290288028a0a9fdfc8fdebebe54a24"},"graph":"https://api.thegraph.com/subgraphs/name/centrifuge/tinlake-kovan-staging","contractConfig":{"JUNIOR_OPERATOR":"ALLOWANCE_OPERATOR","SENIOR_OPERATOR":"PROPORTIONAL_OPERATOR"},"name":"Kovan Naxos Deployment","description":"This is a test description","slug":"testSlug","asset":"invoice"}] 
```

File: config.ts (hunk 7 above)
Comment (ilinzweilin) [reply config.ts:60]: yes, let's kick it out

File: .travis.yml [hunk 12]
```diff
@@ -22,4 +22,4 @@ script:
   - npm run export
 
 env:
```
Comment (philipstanislaus) [inline .travis.yml:24]:
```suggestion
# Make sure you delete spaces behind colons to not confuse yaml (i. e. the JSON should not contain `: `). You can paste the whole `.travis.yml` config here: https://config.travis-ci.com/explore to validate it.
env:
```

File: config.ts (hunk 8 above)
Comment (ilinzweilin) [reply config.ts:61]: yes, totally.
Let's do that when we receive the spec next week

File: components/Loan/List/index.tsx (hunk 1 above)
Comment (ilinzweilin) [reply components/Loan/List/index.tsx:54]: yes it is intentional :) we don;t need it anymore since our current deployments just have one borrower

Comment (ilinzweilin) [reply components/Loan/List/index.tsx:54]: in the demo ui this would be an issue, you are right. We can reintroduce it, once we have a spec for the new demo ui
